### PR TITLE
Active users on top

### PR DESCRIPTION
**Pull request by andpodob**

### CODE REVIEW

File: sticknotes-frontend/src/app/user-list/user-list.component.ts [hunk 1]
```diff
@@ -89,6 +91,8 @@ export class UserListComponent implements OnInit, OnDestroy {
   private fetchActiveUsers(boardId: string): void {
     this.activeUsersService.getActiveUsers(boardId).subscribe(activeUsers => {
       this.activeUsersIdSet = new Set(activeUsers.activeUsers);
+      this.activeUsers = this.usersWithRole.filter(user => this.activeUsersIdSet.has(+user.id));
```
Comment (PaulKovalov) [inline sticknotes-frontend/src/app/user-list/user-list.component.ts:94]: I think its better to use `Number()` to convert string to number

**Comment by AndrzejPodo** [comment]
@googlebot I consent.

File: sticknotes-frontend/src/app/user-list/user-list.component.ts (hunk 1 above)
Comment (AndrzejPodo) [reply sticknotes-frontend/src/app/user-list/user-list.component.ts:94]: why, because of readability or is there any other reason?

Comment (PaulKovalov) [reply sticknotes-frontend/src/app/user-list/user-list.component.ts:94]: yes, basically adding + is JavaScript-ish way to convert string to number, but here we use Typescript, and it has more fancy mechanisms

**Comment by PaulKovalov** [comment]
@AndrzejPodo I've changed base branch there and back to help github recalculate diff

File: sticknotes-frontend/src/app/user-list/user-list.component.html [hunk 2]
```diff
@@ -3,27 +3,46 @@
...
+            {{ userWithRole.role }} <span class="mat-small" style="color:green">ACTIVE</span> 
           </p>
         </div>
         <button mat-icon-button *ngIf="canDelete(userWithRole)" color="primary" (click)="removeUser(userWithRole, $event)"
           aria-label="Open add user dialog">
           <mat-icon>delete</mat-icon>
         </button>
       </mat-list-item>
+
+      <mat-list-item *ngFor="let userWithRole of inactiveUsers" (click)="openEditUserDialog(userWithRole)"
```
Comment (PaulKovalov) [inline sticknotes-frontend/src/app/user-list/user-list.component.html:23]: Active users will be shown in a separate tab?

File: sticknotes-frontend/src/app/user-list/user-list.component.html [hunk 3]
```diff
@@ -3,27 +3,46 @@
...
+          {{ userWithRole.user.email }} 
+          <p class="mat-small"> 
+            {{ userWithRole.role }}
+          </p>
+        </div>
+        <button mat-icon-button *ngIf="canDelete(userWithRole)" color="primary" (click)="removeUser(userWithRole, $event)"
+          aria-label="Open add user dialog">
+          <mat-icon>delete</mat-icon>
+        </button>
+      </mat-list-item>
```
Comment (PaulKovalov) [inline sticknotes-frontend/src/app/user-list/user-list.component.html:37]: all this is duplicated code right?

File: sticknotes-frontend/src/app/user-list/user-list.component.html (hunk 2 above)
Comment (AndrzejPodo) [reply sticknotes-frontend/src/app/user-list/user-list.component.html:23]: No.

File: sticknotes-frontend/src/app/user-list/user-list.component.html (hunk 3 above)
Comment (AndrzejPodo) [reply sticknotes-frontend/src/app/user-list/user-list.component.html:37]: Yes.

Comment (AndrzejPodo) [reply sticknotes-frontend/src/app/user-list/user-list.component.html:37]: We can build component for that, but don't you think it's a bit not necessary?

File: sticknotes-frontend/src/app/user-list/user-list.component.html (hunk 2 above)
Comment (AndrzejPodo) [reply sticknotes-frontend/src/app/user-list/user-list.component.html:23]: List of active users are simply above inactive, this are disjointed sets, right?

File: sticknotes-frontend/src/app/user-list/user-list.component.ts (hunk 1 above)
Comment (AndrzejPodo) [reply sticknotes-frontend/src/app/user-list/user-list.component.ts:94]: Afaik + and Numer() work the same way in terms of input/output, so unary plus operator should be fine. If you really insist we can change it, but imo + looks better as it gives less code.

File: sticknotes-frontend/src/app/user-list/user-list.component.html (hunk 3 above)
Comment (AndrzejPodo) [reply sticknotes-frontend/src/app/user-list/user-list.component.html:37]: Btw. I don't know why, but before this solution I was just sorting users using custom comparator, but changes is order haven't reflected in the UI.

Comment (PaulKovalov) [reply sticknotes-frontend/src/app/user-list/user-list.component.html:37]: I think it is not the case of "unnecessary code".

Comment (PaulKovalov) [reply sticknotes-frontend/src/app/user-list/user-list.component.html:37]: Actually sorting is what I wanted to advise. I think we can make it work with custom comparator

File: sticknotes-frontend/src/app/user-list/user-list.component.ts (hunk 1 above)
Comment (PaulKovalov) [reply sticknotes-frontend/src/app/user-list/user-list.component.ts:94]: I think it is not the case of "unnecessary code".

File: sticknotes-frontend/src/app/user-list/user-list.component.html (hunk 3 above)
Comment (AndrzejPodo) [reply sticknotes-frontend/src/app/user-list/user-list.component.html:37]: I make it using custom comparator but changes in sorting haven't reflected in UI

Comment (PaulKovalov) [reply sticknotes-frontend/src/app/user-list/user-list.component.html:37]: Can you implement this solution please? We'll debug it

File: sticknotes-frontend/src/app/user-list/user-list.component.ts (hunk 1 above)
Comment (AndrzejPodo) [reply sticknotes-frontend/src/app/user-list/user-list.component.ts:94]: ok i can change to Number() :D

Comment (PaulKovalov) [reply sticknotes-frontend/src/app/user-list/user-list.component.ts:94]: thanks!

File: sticknotes-frontend/src/app/user-list/user-list.component.ts [hunk 4]
```diff
@@ -82,13 +84,33 @@ export class UserListComponent implements OnInit, OnDestroy {
...
+   * 
+   * @param activeIdSet 
+   * with a given activeIdSet returns comparator function that:
+   * decides a < b if a is owner or is active and b is not active
+   * decides b > a if b is owner or is active and a is not active
+   * decides a == b in any other case
+   */
+  private compUsers(activeIdSet){
+    return function(a, b){
+      if(a.role === UserRole.OWNER || (activeIdSet.has(+a.id) && !activeIdSet.has(Number(b.id)))){
```
Comment (PaulKovalov) [inline sticknotes-frontend/src/app/user-list/user-list.component.ts:97]: Looks like you decided to mix + with Number :D

Comment (AndrzejPodo) [reply sticknotes-frontend/src/app/user-list/user-list.component.ts:97]: Omg, I haven't noticed that xdd

File: sticknotes-frontend/src/app/user-list/user-list.component.ts [hunk 5]
```diff
@@ -24,6 +24,8 @@ import { ActiveUsersApiService } from '../services/active-users-api.service';
 })
 export class UserListComponent implements OnInit, OnDestroy {
   public usersWithRole: UserBoardRole[] = [];
+  public activeUsers: UserBoardRole[] = [];
+  public inactiveUsers: UserBoardRole[] = [];
```
Comment (PaulKovalov) [inline sticknotes-frontend/src/app/user-list/user-list.component.ts:28]: this is not used

**Review by PaulKovalov — CHANGES_REQUESTED** [review]

**Review by PaulKovalov — APPROVED** [review]

**Pull request merged**